### PR TITLE
update paperweight

### PR DESCRIPTION
**Pull request by amadeusmz**
## Description

Fixes some NMSAPI issues
I've temporary removed some fields/methods that are no longer exists in NMS

## Submitter Checklist

- [x] Make sure you are opening from a topic branch (**/feature/fix/docs/ branch** (right side)) and not your main branch.
- [x] Ensure that the pull request title represents the desired changelog entry.
- [x] New public fields and methods are annotated with `@since TODO`.
- [x] I read and followed the [contribution guidelines](https://github.com/IntellectualSites/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: worldedit-bukkit/adapters/adapter-1_19/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_19_R1/PaperweightPlatformAdapter.java [hunk 1]
```diff
@@ -590,10 +590,11 @@ static void removeBeacon(BlockEntity beacon, LevelChunk levelChunk) {
...
 
     static List<Entity> getEntities(LevelChunk chunk) {
-        if (PaperLib.isPaper()) {
-            return Arrays.asList(chunk.entities.getRawData());
-        }
-        return chunk.level.entityManager.getEntities(chunk.getPos());
+//        if (PaperLib.isPaper()) {
+//            return Arrays.asList(chunk.getBlockEntities().get(0));
+//        }
+//        return chunk.level.enti.getEntities(chunk.getPos());
```
Comment (NotMyFault) [inline worldedit-bukkit/adapters/adapter-1_19/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_19_R1/PaperweightPlatformAdapter.java:596]: This doesn't look right to me

File: worldedit-bukkit/adapters/adapter-1_19/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_19_R1/PaperweightStarlightRelighter.java [hunk 2]
```diff
@@ -8,11 +8,11 @@
 import com.fastasyncworldedit.core.util.MathMan;
 import com.fastasyncworldedit.core.util.TaskManager;
 import com.sk89q.worldedit.internal.util.LogManagerCompat;
+import io.papermc.paper.util.MCUtil;
 import it.unimi.dsi.fastutil.longs.Long2ObjectLinkedOpenHashMap;
 import it.unimi.dsi.fastutil.longs.LongArraySet;
 import it.unimi.dsi.fastutil.longs.LongIterator;
 import it.unimi.dsi.fastutil.longs.LongSet;
-import net.minecraft.server.MCUtil;
```
Comment (NotMyFault) [inline worldedit-bukkit/adapters/adapter-1_19/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_19_R1/PaperweightStarlightRelighter.java:15]: Unused change?

File: worldedit-bukkit/adapters/adapter-1_19/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_19_R1/PaperweightPlatformAdapter.java (hunk 1 above)
Comment (amadeusmz) [reply worldedit-bukkit/adapters/adapter-1_19/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_19_R1/PaperweightPlatformAdapter.java]: The chunk.entities and level.entityManager fields are removed, I was trying to find alternative field and forgot to undo it before commenting.
I fixed it.

File: worldedit-bukkit/adapters/adapter-1_19/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_19_R1/PaperweightStarlightRelighter.java (hunk 2 above)
Comment (amadeusmz) [reply worldedit-bukkit/adapters/adapter-1_19/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_19_R1/PaperweightStarlightRelighter.java:15]: The MCUtil class is moved to paper's utils package.

File: worldedit-bukkit/adapters/adapter-1_19/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_19_R1/PaperweightFaweAdapter.java [hunk 3]
```diff
@@ -490,7 +490,7 @@ public <B extends BlockStateHolder<B>> BlockData adapt(B state) {
     public void sendFakeChunk(org.bukkit.World world, Player player, ChunkPacket chunkPacket) {
         ServerLevel nmsWorld = ((CraftWorld) world).getHandle();
         ChunkHolder map = PaperweightPlatformAdapter.getPlayerChunk(nmsWorld, chunkPacket.getChunkX(), chunkPacket.getChunkZ());
-        if (map != null && map.wasAccessibleSinceLastSave()) {
+        if (map != null) {
```
Comment (SirYwell) [inline worldedit-bukkit/adapters/adapter-1_19/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_19_R1/PaperweightFaweAdapter.java:493]: Does this still behave correctly with the check removed? Does it also still behave correctly on previous versions where this method was available (and likely spigot too)?

File: worldedit-bukkit/adapters/adapter-1_19/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_19_R1/PaperweightPlatformAdapter.java [hunk 4]
```diff
@@ -590,10 +590,11 @@ static void removeBeacon(BlockEntity beacon, LevelChunk levelChunk) {
...
     static List<Entity> getEntities(LevelChunk chunk) {
-        if (PaperLib.isPaper()) {
-            return Arrays.asList(chunk.entities.getRawData());
-        }
-        return chunk.level.entityManager.getEntities(chunk.getPos());
+//        if (PaperLib.isPaper()) {
+//            return Arrays.asList(chunk.entities.getRawData());
+//        }
+//        return chunk.level.entityManager.getEntities(chunk.getPos());
+        return new ArrayList<>();
```
Comment (SirYwell) [inline worldedit-bukkit/adapters/adapter-1_19/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_19_R1/PaperweightPlatformAdapter.java:597]: This method needs to support:
- Spigot
- Paper before the new chunk system
- current Paper

File: worldedit-bukkit/adapters/adapter-1_19/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_19_R1/PaperweightStarlightRelighter.java (hunk 2 above)
Comment (SirYwell) [reply worldedit-bukkit/adapters/adapter-1_19/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_19_R1/PaperweightStarlightRelighter.java:15]: This needs support for older Paper builds too, as we don't have separate implementations for 1.19/1.19.1 and 1.19.2

**Review by SirYwell — CHANGES_REQUESTED** [review]

**Pull request closed without merging**